### DISCUSSION TITLE
MIDI custom project compiling problem

### DISCUSSION THREAD

**Discussion by icyrogue** [hathach/tinyusb#1524] [Q&A]
Hey there! 

When compiling my MIDI tinyusb example code in my own project I got this error:

`
/Users/icy/pico-sdk/lib/tinyusb/src/portable/raspberrypi/rp2040/dcd_rp2040.c: In function 'dcd_edpt_open':
/Users/icy/pico-sdk/lib/tinyusb/src/portable/raspberrypi/rp2040/dcd_rp2040.c:429:76: error: request for member 'size' in something not a structure or union
  429 |     hw_endpoint_init(desc_edpt->bEndpointAddress, desc_edpt->wMaxPacketSize.size, desc_edpt->bmAttributes.xfer);
      |                                                                            ^
make[2]: *** [CMakeFiles/main.dir/Users/icy/pico-sdk/lib/tinyusb/src/portable/raspberrypi/rp2040/dcd_rp2040.c.obj] Error 1
make[1]: *** [CMakeFiles/main.dir/all] Error 2
make: *** [all] Error 2
`

I'm guessing that problem is in CmakeLists file since MIDI example from the repo works perfectly for me, but I'm also out of ideas of how to fix that 

My CmakeLists.txt for your inspection:

```
cmake_minimum_required(VERSION 3.12)

# Pull in PICO SDK (must be before project)
include(pico_sdk_import.cmake)

project(main C CXX ASM)
set(CMAKE_C_STANDARD 11)
set(CMAKE_CXX_STANDARD 17)

set(PICO_EXAMPLES_PATH ${PROJECT_SOURCE_DIR})

# Initialize the SDK
pico_sdk_init()

add_executable(main
        main.c
        )

        target_include_directories(${PROJECT_NAME} PUBLIC
        ${PROJECT_SOURCE_DIR}/tinyusb/src
        ${CMAKE_CURRENT_LIST_DIR}

      )

      target_sources(${PROJECT_NAME} PUBLIC
      ${CMAKE_CURRENT_LIST_DIR}/main.c
      ${CMAKE_CURRENT_LIST_DIR}/usb_descriptors.c
      )

# Pull in our pico_stdlib which aggregates commonly used features
target_link_libraries(main pico_stdlib pico_bootrom tinyusb_device tinyusb_board)

# create map/bin/hex/uf2 file etc.
pico_add_extra_outputs(main)
```
My project tree:

```
main
├── build
│  ├── cmake_install.cmake
│  ├── CMakeCache.txt
│  ├── CMakeFiles
│  ├── elf2uf2
│  ├── generated
│  ├── Makefile
│  └── pico-sdk
├── CMakeLists.txt
├── hw
│  ├── bsp
│  └── mcu
├── main.c
├── pico_sdk_import.cmake
├── tinyusb
│  ├── CODE_OF_CONDUCT.rst
│  ├── CONTRIBUTORS.rst
│  ├── docs
│  ├── examples
│  ├── hw
│  ├── lib
│  ├── LICENSE
│  ├── pkg.yml
│  ├── README.rst
│  ├── repository.yml
│  ├── src
│  ├── test
│  ├── tools
│  └── version.yml
├── tusb.c
├── tusb_config.h
└── usb_descriptors.c
```
main.c is a copy of MIDI example 

Thanks for all the help in advance

**Comment by hathach** [comment]
instead explaining everything in text, you should create an github project for others to comment on. Anyway, I am not good with cmake as well, you should ask this on pico forum there is more rpi team member there to help you out.
